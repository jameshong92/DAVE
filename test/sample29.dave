// test table access
int main() {
    rec r1 = new rec(age: 22, name:"Yang", GPA:3.50); 
    rec r2 = new rec(name:"James", age:23, GPA:3.88);
    rec r3 = new rec(age:24, GPA:3.80, name:"Min Woo");
    tbl test = tbl([r1, r2, r3]);
    print(test);
    string temp1[] = ["age", "name"];
    tbl test1 = access(test, temp1, getArrayLen(temp1));
    print(test1);
    tbl test2 = access(test, ["name", "gender", "tall"]);
    print(test2);
}