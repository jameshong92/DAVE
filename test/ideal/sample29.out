age	name	GPA	
22	Yang	3.50	
23	James	3.88	
24	Min Woo	3.80	

age	name	
22	Yang	
23	James	
24	Min Woo	

name	
Yang	
James	
Min Woo	

